orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/model/support/TriggerDeserializer.kt

```diff
@@ -42,6 +42,12 @@ class TriggerDeserializer :
   override fun deserialize(parser: JsonParser, context: DeserializationContext): Trigger =
     parser.codec.readTree<JsonNode>(parser).run {
       return when {
+        looksLikeCustom() -> {
```

Comment:
Why did we move this? For priority?

Hmm, that could cause a lot of issues that may not be obvious. I imagine custom should be one of the least priority-wise deserializers. Why did we make it first?

I think we would need to run this in a more robust environment. Unfortunately Im not comfortable in our current tests that might ensure this will not be a breaking change. Can you give me reasons on why this needed to be moved?

Reply:
I moved it at the beginning for priority. As I am defining a custom deserializer that will override the PipelineTrigger If it is not at the beginning the deserializer won't convert a PipelineTrigger into a PipelineRefTrigger.